core/file.c

```diff
@@ -428,20 +428,15 @@ static int parse_file_buffer(const char *filename, struct memblock *mem)
 	return parse_xml_buffer(filename, mem->buffer, mem->size, &dive_table, NULL);
 }
 
-int check_git_sha(const char *filename, struct git_repository **git_p, const char **branch_p)
+int check_git_sha(const char *loc, const char *branch, const char *user, bool is_remote, bool is_cloud, struct git_repository **git_p)
 {
 	struct git_repository *git;
-	const char *branch = NULL;
 
 	char *current_sha = strdup(saved_git_id);
-	git = is_git_repository(filename, &branch, NULL, false);
+	git = is_git_repository(loc, branch, user, is_remote, is_cloud);
 	if (git_p)
 		*git_p = git;
-	if (branch_p)
-		*branch_p = branch;
-	if (prefs.cloud_git_url &&
-	    strstr(filename, prefs.cloud_git_url)
-	    && git == dummy_git_repository) {
+	if (is_cloud && git == dummy_git_repository) {
```

Comment:
This changes a ton of things at once which makes it hard to comment on the commit.
it looks like branch changed from an output parameter to an input parameter?

Without more comments I have no idea what this is doing, tbh

Reply:
Yes indeed - the parsing of URLs is now done _outside_ of the git machinery, so branch is not an out parameter anymore. In the next commit, a `struct git_state` is introduced which describes everything that is needed to address the repo (repository, branch, username) and some state (number of authorization attempts, current sha).